Source/dthread.cpp

```diff
@@ -80,7 +80,7 @@ void dthread_send_delta(int pnum, _cmd_id cmd, byte *pbSrc, int dwLen)
 		return;
 	}
 
-	pkt = static_cast<TMegaPkt *>(std::malloc(dwLen + 20));
+	pkt = reinterpret_cast<TMegaPkt *>(new byte[dwLen + 20]);

```

Comment:
This results in calling `delete` on a pointer allocated with `new[]`, I think this is undefined behaviour? Allocating a buffer smaller than the struct size might already be in that realm anyway so it's likely no worse than the current implementation 😅

Reply:
I have some lingering doubts about the UB stuff as well.

I did take a look at what gcc outputs in cases like these and it seems to be correct.

Reply:
Might as well cast it back to byte[] and then call delete[] :laughing:

Reply:
Same comment as below applies to this (needs to be an aligned `new []` and a corresponding aligned `delete []`)